src/main/java/com/github/brainlag/nsq/lookup/DefaultNSQLookup.java

```diff
@@ -26,16 +27,29 @@ public void addLookupAddress(String addr, int port) {
         Set<ServerAddress> addresses = Sets.newHashSet();
 
         for (String addr : this.addresses) {
-            ObjectMapper mapper = new ObjectMapper();
-
-            JsonNode jsonNode = mapper.readTree(new URL(addr + "/lookup?topic=" + topic));
-            JsonNode producers = jsonNode.get("data").get("producers");
-            for (JsonNode node : producers) {
-                String host = node.get("broadcast_address").asText();
-                ServerAddress address = new ServerAddress(host, node.get("tcp_port").asInt());
-                addresses.add(address);
+            try {
+                ObjectMapper mapper = new ObjectMapper();
+
+                JsonNode jsonNode = mapper.readTree(new URL(addr + "/lookup?topic=" + topic));
+                LogManager.getLogger(this).warn("Server connection information: " + jsonNode.toString());
+                JsonNode producers = jsonNode.get("data").get("producers");
+                for (JsonNode node : producers) {
+                    String host = node.get("broadcast_address").asText();
+                    ServerAddress address = new ServerAddress(host, node.get("tcp_port").asInt());
+                    addresses.add(address);
+                }
+            } catch (IOException e) {
+                LogManager.getLogger(this).warn("Unable to connect to address " + addr);
             }
         }
+        if (addresses.isEmpty()) {
+            throw new IOException("Unable to connect to any NSQ Lookup servers, servers tried: " + this.addresses.toString());
```

Comment:
If addresses is empty then there are no NSQ Lookup servers configured. Therefore it is not an IOException. 
We should probably not throw an exception at all and just log an error message.